### ISSUE TITLE
Check color contrast and provide recommendations to meet WCAG guidelines

### ISSUE THREAD

**Issue by bemdesign-er** [WordPress/gutenberg#16826]
**Is your feature request related to a problem? Please describe.**
See https://github.com/WordPress/gutenberg/issues/16732 and design triage discussion here: https://wordpress.slack.com/archives/C02S78ZAL/p1564502856190700. When switching to dark mode or high contrast mode, there should be a way for the system to automatically create appropriate color-contrast styling.

**Describe the solution you'd like**
Ideally there would be a complete in-system color contrast checker that could do 2 things: 1) automatically create appropriate color contrast styling for editor elements (for Dark Mode or High Contrast modes), and 2) test user-selected colors and provide color combination recommendations to improve accessibility (for use in text + background combination components).

**Comment by aristath** [comment]
Just to add here, a more abstract method to detect and define color-schemes would be preferable.
We currently have light and dark themes, but there's also the `prefers-color-scheme` media-query that themes may implement. For that we'd also need a way to detect and apply color-scheme changes.
Ties into https://github.com/WordPress/gutenberg/issues/24368
